numpy/random/mtrand/mtrand.pyx

```diff
@@ -973,7 +973,7 @@ cdef class RandomState:
             raise ValueError("low is out of bounds for %s" % (key,))
         if high > highbnd:
             raise ValueError("high is out of bounds for %s" % (key,))
-        if low >= high:
+        if low >= high and np.prod(size) != 0:
             raise ValueError("low >= high")
```

Comment:
This should probably become `"Range cannot be empty (low >= high) unless 0 samples are taken"`